---
title: "ASM treatment exp"
author: "Rose kithan"
date: "9/18/2022"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = "~/Documents/Sundin_Lab/Rot_microbiome/") #set working directory
```


# Load packages
load R packages and custom functions 
```{r}
library("phyloseq")
library("ggplot2")
library("ape")
library("gridExtra")
library("ggpubr")
library("ggsignif")
library(forcats)
library(vegan)
library(cowplot)
library(tidyverse)
library(agricolae)
theme_set(theme_cowplot())
set.seed(7)
source("C:/users/kithanro/Documents/16s_Gala_postharvest_2021/DenefLab-MicrobeMiseq/R/miseqR.R")
```

```{r}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = "C:/Users/kithanro/Documents/16s_Gala_postharvest_2021")
```


#Phyloseq object
```{r echo=TRUE}
#read in otu table
otu_table <- read.table("OTU_table.txt",
                        sep="\t",row.names = 1,header = TRUE) %>%
  as.matrix(otu_table,rownames=TRUE)

#read in taxonomy
taxonomy <- read_delim("taxonomy.csv", delim = ",", col_names = TRUE) %>% 
  column_to_rownames(var = "OTU_ID") %>%
  as.matrix()

#read in metadata
metadata <- read.table("Sample_metadata.tsv", sep = "\t", header = TRUE,
                    row.names=1, na.strings = "")  %>%
  mutate(Timepoints = factor(Timepoints, ordered = TRUE, levels = c("Day 0", "Day 7", "Day 14", "Day 25", "Day 49")))
metadata
phy_tree <- read_tree("tree.nwk")

#import as phyloseq objects
OTU = otu_table(otu_table,taxa_are_rows=TRUE)
TAX = tax_table(taxonomy)
META = sample_data(metadata)
#Physeq object
physeq <- phyloseq(OTU,TAX,META,phy_tree) #1796 taxa and 174 samples
```

#Fitering Chloroplast and Mitocondria OTUs
```{r}
#physeq <- physeq_filtered
physeq_filtered <- physeq %>% 
  
  subset_taxa(Order != " Chloroplast") %>%
  subset_taxa(Family != " Mitohondria") #1469 taxa and 174 samples 

#check reads per sample
test_df <- as.data.frame(sample_sums(physeq_filtered)) %>% #sum reads per sample
  rownames_to_column(var = "Sample_ID") #format for plot

# make the plot
test_plot <- ggplot(test_df, aes(x = reorder(Sample_ID, -`sample_sums(physeq_filtered)`), #orders axis
                                 y = `sample_sums(physeq_filtered)`
                                 )) + 
  geom_col() + ylab("Reads per sample") + xlab("Sample_ID") + #clean up labels
  theme(axis.text.x = element_text(angle = -90, hjust=0))
test_plot

# more filtering to remove samples that didn't sequence well or OTUs that may not be real 
  physeq_filtered2 <- physeq_filtered %>%  
  prune_samples(sample_sums(.) >= 200, .) #remove samples with <=200 reads per samples    #1469 taxa and 148 samples
  #prune_taxa(taxa_sums(.) >= 3, .) # remove taxa with <=5 reads per OTU
  
# subset samples to each read numbers 
  
#extrct sample df from phylose object
test_samp <- sample_data(physeq_filtered2)
```


# Import data
### Shared 
import shared (Otu) table and format to make phyloseq object 
```{r}
shared <- read_delim(file = "RotOTUs.shared", delim = "\t", 
                       col_names = TRUE) %>%
  select(-label, -numOtus) %>% #drop extra columns from mothur 
  column_to_rownames(var = "Group") %>%
  otu_table(taxa_are_rows = FALSE) #make phlyloseq class object 

#taxa_names(shared) #these should always be OTUs
#sample_names(shared) #should match samples in sequence file 
```
### Taxa
import taxonomy table and format to make phyloseq object 
```{r}
tax <- read_delim(file = "RotOTUs.taxonomy_edited.csv", delim = ",", 
                       col_names = TRUE) %>%
  select(-"Size", -"D", -"P", -"C", -"O", -"F", -"G") %>% #drop extra columns from mothur 
  #as.matrix(rownames = "OTU") %>%
  column_to_rownames(var = "OTU") %>%
  as.matrix() %>%
  tax_table() #make phlyloseq class object 

#taxa_names(shared) #these are OTUs
```
### Metadata 
sample metadata
```{r}
metadata <- read_delim(file = "Sample_metadata_file_16S.csv", delim = ",", 
                       col_names = TRUE) %>%
  filter(Cultivar != "Cherry",
         Cultivar != "Mock") %>% #remove Kristi's samples 
  mutate(Sample_labels = Sample_ID,
         Trt_Tiss = base::paste(Treatment, Tissue, sep = "-"), #create extra category 
         Timepoint = recode(Timepoint, #relabel days for easy reading 
                             "1"= "Day 0",
                             "2" = "Day 14",
                             "3" = "Day 28",
                             "4" = "Day 42",
                             "5" = "Day 56"),
         Timepoint = factor(Timepoint, ordered = TRUE,
                            levels = c("Day 0", "Day 14", "Day 28",
                                       "Day 42", "Day 56", "Rotten"))) %>% #set day order 
  column_to_rownames(var = "Sample_ID")  %>%
  sample_data() #make phlyloseq class object 

#sample_names(shared)
```

# Phyloseq object
Merge previous objects into a phyloseq object 
```{r}
psq <- phyloseq(metadata, shared, tax)
psq
#saveRDS(psq, file = "psq_object.RDS", compress = TRUE) #save R object 
```
Make phyloseq object into data frame
```{r eval=FALSE, include=FALSE}
psq_lng <- psq %>%
  #tax_glom() %>%
  psmelt() %>%
  group_by() %>% 
  summarise() %>% 
  filter()
```

# Richness in Pulp. Gala Vs. Jonathan at different time points

```{r fig.width=10, fig.height=40}
alpha_diversity <- estimate_richness(physeq_filtered2, measures = c("Shannon", "Chao1"))
df_alpha <- data.frame(alpha_diversity, sample_data(physeq_filtered2))
df <- reshape2::melt(df_alpha,
                     measure.var=c("Shannon", "Chao1"),
                     id.vars=c("Timepoints", "Cultivar", "Tissue_types", "Treatments"))
df2 <- df %>%
  #mutate(value = as.numeric(value)) %>% 
  #as.numeric(df$value)
  filter(Treatments != "None") %>%
  #filter(Tissue.types != "Skin") %>%
  # filter(Timepoints != "Day 14")  %>%
  # filter(Timepoints != "Day 49")  %>%
  # filter(Timepoints != "Day 25")  %>%
  # filter(Timepoints != "Day 7")  %>%
  # filter(Tissue_types == "Pulp")  %>%
  filter(Timepoints == "Day 49") %>%
  filter(Treatments != "Control") %>%
  #filter(Tissue_types == "Skin") %>%
  filter(Treatments == "ASM-treated")  
   
 
#selecting the rows corresponding to each index from the main dataframe
shannon = dplyr::filter(df2, variable == "Shannon")
chao = dplyr::filter(df2, variable == "Chao1")
```

```{r}
# conduct ANOVA
total_aov <- df2 %>%
  filter(variable == "Shannon") %>%
  aov(value ~ Tissue_types, data = .)
summary(total_aov) # p = .98

# # conduct Tukey's HSD test (this is extra)
# area_tuk <- tidy(TukeyHSD(total_aov)) %>%
#   mutate(adj.p.value = round(adj.p.value, digits = 4)) %>%
#   filter(adj.p.value <= 0.1000) 

# conduct Tukey's HSD test with agricolae and get letters for figure 
#area_let <- as.matrix(HSD.test(total_aov,"Timepoints", group=TRUE, alpha = 0.05)$groups) %>%
  area_let <- as.matrix(HSD.test(total_aov,"Tissue_types", group=TRUE)$groups) %>%
  as.data.frame() %>%                       #formatting the table with the letters
  rownames_to_column(var="Tissue_types") %>%
  mutate(value = round(as.numeric(as.character(value)), digits=2),
         groups = as.character(groups)) 

# plot it 

plot_total <- df2 %>%
  filter(variable == "Shannon") %>%
  ggplot(aes(x = Tissue_types, y = value, fill = Cultivar)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(color="black", size=1, alpha=0.5, 
             position = position_jitterdodge(jitter.width = 0.1)) +
  facet_grid(variable~., scales = "free") +
  labs(y="Alpha Diversity Index", x="Tissue_types") +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
  theme_classic() + 
  theme(text = element_text(size = 15)) +
  theme(axis.text = element_text(size = 15)) +
  theme(axis.title.x=element_blank()) + 
  ggtitle("ASM treated- Skin vs. Pulp. Day 49") +
   scale_fill_aaas() +
  annotate("text", x=area_let$Tissue_types, y = 3.2, size = 6, label=area_let$groups) 
plot_total
```


# Richness in Control: Skin vs. pulp day 0

```{r fig.width=10, fig.height=40}
alpha_diversity <- estimate_richness(physeq_filtered2, measures = c("Shannon", "Chao1"))
df_alpha <- data.frame(alpha_diversity, sample_data(physeq_filtered2))
df <- reshape2::melt(df_alpha,
                     measure.var=c("Shannon", "Chao1"),
                     id.vars=c("Timepoints", "Cultivar", "Tissue_types", "Treatments"))
df2 <- df %>%
  #mutate(value = as.numeric(value)) %>% 
  #as.numeric(df$value)
  filter(Treatments != "None") %>%
  #filter(Tissue.types != "Skin") %>%
  # filter(Timepoints != "Day 14")  %>%
  # filter(Timepoints != "Day 49")  %>%
  # filter(Timepoints != "Day 25")  %>%
  # filter(Timepoints != "Day 7")  %>%
  # filter(Tissue_types == "Pulp")  %>%
  filter(Timepoints == "Day 0") %>%
  #filter(Treatments != "Control") %>%
  #filter(Tissue_types == "Skin") %>%
  filter(Treatments == "Control")  
   
 
#selecting the rows corresponding to each index from the main dataframe
shannon = dplyr::filter(df2, variable == "Shannon")
chao = dplyr::filter(df2, variable == "Chao1")
```

```{r}
# conduct ANOVA
total_aov <- df2 %>%
  filter(variable == "Shannon") %>%
  aov(value ~ Tissue_types, data = .)
summary(total_aov) # p = .98

# # conduct Tukey's HSD test (this is extra)
# area_tuk <- tidy(TukeyHSD(total_aov)) %>%
#   mutate(adj.p.value = round(adj.p.value, digits = 4)) %>%
#   filter(adj.p.value <= 0.1000) 

# conduct Tukey's HSD test with agricolae and get letters for figure 
#area_let <- as.matrix(HSD.test(total_aov,"Timepoints", group=TRUE, alpha = 0.05)$groups) %>%
  area_let <- as.matrix(HSD.test(total_aov,"Tissue_types", group=TRUE)$groups) %>%
  as.data.frame() %>%                       #formatting the table with the letters
  rownames_to_column(var="Tissue_types") %>%
  mutate(value = round(as.numeric(as.character(value)), digits=2),
         groups = as.character(groups)) 

# plot it 

plot_total <- df2 %>%
  filter(variable == "Shannon") %>%
  ggplot(aes(x = Tissue_types, y = value, fill = Cultivar)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(color="black", size=1, alpha=0.5, 
             position = position_jitterdodge(jitter.width = 0.1)) +
  facet_grid(variable~., scales = "free") +
  labs(y="Alpha Diversity Index", x="Tissue_types") +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
  theme_classic() + 
  theme(text = element_text(size = 15)) +
  theme(axis.text = element_text(size = 15)) +
  theme(axis.title.x=element_blank()) + 
  ggtitle("Control- Skin vs. Pulp. Day 0") +
   scale_fill_aaas() +
  annotate("text", x=area_let$Tissue_types, y = 3.2, size = 6, label=area_let$groups) 
plot_total
```



# Richness in Control vs. ASM-treated. Pulp day 0

```{r fig.width=10, fig.height=40}
alpha_diversity <- estimate_richness(physeq_filtered2, measures = c("Shannon", "Chao1"))
df_alpha <- data.frame(alpha_diversity, sample_data(physeq_filtered2))
df <- reshape2::melt(df_alpha,
                     measure.var=c("Shannon", "Chao1"),
                     id.vars=c("Timepoints", "Cultivar", "Tissue_types", "Treatments"))
df2 <- df %>%
  #mutate(value = as.numeric(value)) %>% 
  #as.numeric(df$value)
  filter(Treatments != "None") %>%
  #filter(Tissue.types != "Skin") %>%
  # filter(Timepoints != "Day 14")  %>%
  # filter(Timepoints != "Day 49")  %>%
  # filter(Timepoints != "Day 25")  %>%
  # filter(Timepoints != "Day 7")  %>%
  filter(Tissue_types == "Skin")  %>%
  filter(Timepoints == "Day 7") 
  #filter(Treatments != "Control") %>%
  #filter(Tissue_types == "Skin") %>%
  #filter(Treatments == "Control")  
   
 
#selecting the rows corresponding to each index from the main dataframe
shannon = dplyr::filter(df2, variable == "Shannon")
chao = dplyr::filter(df2, variable == "Chao1")
```

```{r}
# conduct ANOVA
total_aov <- df2 %>%
  filter(variable == "Shannon") %>%
  aov(value ~ Treatments, data = .)
summary(total_aov) # p = .98

# # conduct Tukey's HSD test (this is extra)
# area_tuk <- tidy(TukeyHSD(total_aov)) %>%
#   mutate(adj.p.value = round(adj.p.value, digits = 4)) %>%
#   filter(adj.p.value <= 0.1000) 

# conduct Tukey's HSD test with agricolae and get letters for figure 
#area_let <- as.matrix(HSD.test(total_aov,"Timepoints", group=TRUE, alpha = 0.05)$groups) %>%
  area_let <- as.matrix(HSD.test(total_aov,"Treatments", group=TRUE)$groups) %>%
  as.data.frame() %>%                       #formatting the table with the letters
  rownames_to_column(var="Treatments") %>%
  mutate(value = round(as.numeric(as.character(value)), digits=2),
         groups = as.character(groups)) 

# plot it 

plot_total <- df2 %>%
  filter(variable == "Shannon") %>%
  ggplot(aes(x = Treatments, y = value, fill = Cultivar)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(color="black", size=1, alpha=0.5, 
             position = position_jitterdodge(jitter.width = 0.1)) +
  facet_grid(variable~., scales = "free") +
  labs(y="Alpha Diversity Index", x="Treatments") +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
  theme_classic() + 
  theme(text = element_text(size = 15)) +
  theme(axis.text = element_text(size = 15)) +
  theme(axis.title.x=element_blank()) + 
  ggtitle("Control Vs. ASM-treated. Skin: Day 7") +
   scale_fill_aaas() +
  annotate("text", x=area_let$Treatments, y = 3.2, size = 6, label=area_let$groups) 
plot_total
```













# Richness
### Shelf life
Subset metadata for untreated samples
```{r}
untreat_data <- read_delim(file = "Sample_metadata_file_16S.csv", delim = ",", 
                       col_names = TRUE) %>%
  #distinct(., Sample_ID, .keep_all = TRUE) %>%
  filter(Cultivar != "Cherry",
         Cultivar != "Mock",
         Timepoint != "Rotten",
         Treatment == "Untreated")
```
Calculate alpha diversity metrics, and format data for plotting. 
```{r}
richness_dfu <- psq %>%
  # subset phyloseq object as for sample data
  subset_samples(Cultivar != "Cherry") %>% 
  subset_samples(Cultivar != "Mock") %>%
  subset_samples(Timepoint != "Rotten") %>% 
  subset_samples(Treatment == "Untreated") %>%
  # calculate richness (observed OTUs)
  estimate_richness(., split = TRUE,  measures = c("Observed", "Shannon")) %>% 
  # make sample_id column before join 
  rownames_to_column(var = "Sample_ID") %>% 
  # join with sample meta data from section above
  inner_join(untreat_data, by = "Sample_ID") %>%  
  rename(Richness = Observed) %>%
  # calculate average richness per group 
  group_by(Tissue, Tree, Timepoint, Cultivar, Treatment) %>%
  summarise(Avg_Richness = round(mean(Richness), digits = 0),
            Avg_Shannon = round(mean(Shannon), digits = 2)) %>%
  # relable timepoints and set order 
  mutate(Timepoint = recode(Timepoint,
                             "1"= "Day 0",
                             "2" = "Day 14",
                             "3" = "Day 28",
                             "4" = "Day 42",
                             "5" = "Day 56"),
         Timepoint = factor(Timepoint, ordered = TRUE,
                            levels = c("Day 0", "Day 14", "Day 28",
                                       "Day 42", "Day 56", "Rotten"))) 
```

Generate richness time course plot.
```{r include=FALSE}
richu <- psq %>%
  # subset phyloseq object as for sample data  
  subset_samples(Cultivar != "Cherry") %>%
  subset_samples(Cultivar != "Mock") %>% 
  subset_samples(Timepoint != "Rotten") %>%
  subset_samples(Treatment == "Untreated") %>% 
  # make the plot 
  plot_richness(., "Timepoint", measures = c("Observed")) +
  facet_grid(Cultivar~Tissue) +
  ylab("Richness (Observed ESVs)") + xlab(NULL)
richu
```
Make it pretty. 
```{r}
# lines for error bars
line_posits <- richness_dfu %>% 
  filter(Cultivar != "Cherry",
         Cultivar != "Mock",
         Timepoint != "Rotten",
         Treatment == "Untreated") %>%
  group_by(Cultivar, Tissue, Timepoint) %>%
  summarise(group_mean = mean(Avg_Richness, na.rm = TRUE), 
            group_sd = sd(Avg_Richness, na.rm = TRUE))

#drop first geom_point layer
richu$layers <- richu$layers[-1] 

richU <- richu +   
  # add new points layer in color
  geom_jitter(aes(color = Timepoint), width = 0.3) + 
  # plot standard deviation
  geom_errorbar(data=line_posits, aes(x=Timepoint, 
                                      y=group_mean,  
                                      ymin=group_mean-group_sd, 
                                      ymax=group_mean+group_sd), 
                width=0.1, color = "black") +
  # plot mean line 
  geom_errorbar(data=line_posits, aes(x=Timepoint, 
                                      y=group_mean,  
                                      ymin=group_mean, 
                                      ymax=group_mean),
                width=0.4, color = "black") +
  # change colors 
  scale_color_manual(name = "Timepoint",
                     breaks = c("Day 0", "Day 14", "Day 28", 
                                "Day 42", "Rotten"),
                     values = c("#ED1E24", "#9ACA3C", "#40B9EB",
                                "#426FB6", "#D771AD")) +
  theme(legend.position = "none")
richU
```

### Chitosan
Subset for chitosan treated and control samples
```{r include=FALSE}
chit_data <- read_delim(file = "Sample_metadata_file_16S.csv", delim = ",", 
                       col_names = TRUE) %>%
  #distinct(., Sample_ID, .keep_all = TRUE) %>%
  filter(Cultivar != "Cherry",
         Cultivar != "Mock",
         Timepoint != "Rotten",
         Treatment == "Chitosan" | Treatment == "Mock")
```
Calculate alpha diversity metrics, and format data for plotting. 
```{r}
richness_dfc <- psq %>%
  # subset phyloseq object as for sample data
  subset_samples(Cultivar != "Cherry") %>% 
  subset_samples(Cultivar != "Mock") %>%
  subset_samples(Timepoint != "Rotten") %>% 
  subset_samples(Treatment == "Chitosan" | Treatment == "Mock") %>%
  # calculate richness (observed OTUs)
  estimate_richness(., split = TRUE,  measures = c("Observed", "Shannon")) %>% 
  # make sample_id column before join 
  rownames_to_column(var = "Sample_ID") %>% 
  # join with sample data imported above
  inner_join(chit_data, by = "Sample_ID") %>%  
  rename(Richness = Observed) %>%
  # calculate average richness per group 
  group_by(Tissue, Tree, Timepoint, Cultivar, Treatment) %>%
  summarise(Avg_Richness = round(mean(Richness), digits = 0),
            Avg_Shannon = round(mean(Shannon), digits = 2)) %>%
  mutate(Timepoint = recode(Timepoint,
                             "1"= "Day 0",
                             "2" = "Day 14",
                             "3" = "Day 28",
                             "4" = "Day 42",
                             "5" = "Day 56"),
         Timepoint = factor(Timepoint, ordered = TRUE,
                            levels = c("Day 0", "Day 14", "Day 28",
                                       "Day 42", "Day 56", "Rotten"))) 
```
Generate richness jonathan chitosan plot.
```{r include=FALSE}
richcj <- psq %>%
  # subset phyloseq object as for sample data  
  subset_samples(Timepoint != "Rotten") %>%
  subset_samples(Cultivar == "Jonathan") %>%
  subset_samples(Treatment == "Chitosan" | Treatment == "Mock") %>% 
  # make the plot 
  plot_richness(., "Timepoint", measures = c("Observed")) +
  facet_grid(Treatment~Tissue) + xlab(NULL)
richcj
```
Make it pretty. 
```{r}
# lines for error bars
line_posits <- richness_dfc %>% 
  filter(Cultivar == "Jonathan",
         Timepoint != "Rotten",
         Treatment == "Chitosan" | Treatment == "Mocck") %>%
  group_by(Cultivar, Tissue, Timepoint) %>%
  summarise(group_mean = mean(Avg_Richness, na.rm = TRUE), 
            group_sd = sd(Avg_Richness, na.rm = TRUE))

richcj$layers <- richcj$layers[-1] #drop geom_point 

richCj <- richcj +  
  #add individual points in color
  geom_jitter(aes(color = Timepoint)) +  
  scale_y_continuous(name = "Richness (Observed ESVs)",
                     limits = c(0, 40),
                     breaks = c(0, 10, 20, 30, 40)) + 
  # plot standard deviation
  geom_errorbar(data=line_posits, aes(x=Timepoint, 
                                      y=group_mean,  
                                      ymin=group_mean-group_sd, 
                                      ymax=group_mean+group_sd), 
                width=0.1, color = "black") +
  # plot mean line 
  geom_errorbar(data=line_posits, aes(x=Timepoint, 
                                      y=group_mean,  
                                      ymin=group_mean, 
                                      ymax=group_mean),
                width=0.4, color = "black") + 
  # change colors 
  scale_color_manual(name = "Timepoint",
                     breaks = c("Day 0", "Day 14", "Day 28", 
                                "Day 42", "Rotten"),
                     values = c("#ED1E24", "#9ACA3C", "#40B9EB",
                                "#426FB6", "#D771AD")) +
  theme(legend.position = "none") 
richCj
```

### Rotten
Subset sample data for rotten timepoint
```{r include=FALSE}
rot_data <- read_delim(file = "Sample_metadata_file_16S.csv", delim = ",", 
                       col_names = TRUE) %>%
  filter(Cultivar != "Cherry",
         Cultivar != "Mock",
         Timepoint == "Rotten")
```
Calculate alpha diversity metrics, and format data for plotting. 
```{r}
richness_dfR <- psq %>%
  # subset phyloseq object as for sample data 
  subset_samples(Cultivar != "Cherry") %>% 
  subset_samples(Cultivar != "Mock") %>% 
  subset_samples(Timepoint == "Rotten") %>% 
  # calculate richness (observed OTUs)
  estimate_richness(., split = TRUE,  measures = c("Observed", "Shannon")) %>% 
  # make sample_id column before join 
  rownames_to_column(var = "Sample_ID") %>% 
  # join with sample data imported above
  inner_join(rot_data, by = "Sample_ID") %>%  
  rename(Richness = Observed) %>%
  # calculate average richness per group 
  group_by(Tissue, Tree, Timepoint, Cultivar, Treatment) %>%
  summarise(Avg_Richness = round(mean(Richness), digits = 0),
            Avg_Shannon = round(mean(Shannon), digits = 2)) 
```
Generate rotten richness plot.
```{r}
richr <- psq %>%
  # subset phyloseq object as for sample data  
  subset_samples(Cultivar != "Cherry") %>% 
  subset_samples(Cultivar != "Mock") %>% 
  subset_samples(Timepoint == "Rotten") %>%
  # make the plot 
  plot_richness(., "Cultivar", 
                measures = c("Observed"), 
                color = "Tissue") +
  facet_grid(~Treatment, labeller = labeller(Untreated = "Rotten")) +
  xlab(NULL)
richr
```
Make it pretty. 
```{r}
# lines for error bars
line_posits <- richness_dfR %>% 
  filter(Cultivar != "Cherry",
         Cultivar != "Mock",
         Timepoint == "Rotten") %>%
  group_by(Cultivar, Tissue, Treatment) %>%
  summarise(group_mean = mean(Avg_Richness, na.rm = TRUE), 
            group_sd = sd(Avg_Richness, na.rm = TRUE))

richr$layers <- richr$layers[-1] #drop geom_point 

richR <- richr +   
  #add violin in color: geom_violin(aes(color = Tissue)) + 
  # fill points overtop of violin plots 
  geom_point(aes(color = Tissue), 
             position = position_jitterdodge(jitter.width = NULL,
                                             jitter.height = 0,
                                             dodge.width = 0.95)) + 
  # plot standard deviation
  geom_errorbar(data=line_posits, aes(x=Cultivar, 
                                      y=group_mean,  
                                      ymin=group_mean-group_sd, 
                                      ymax=group_mean+group_sd), 
                width=0.1, color = "black") +
  # plot mean line 
  geom_errorbar(data=line_posits, aes(x=Cultivar, 
                                      y=group_mean,  
                                      ymin=group_mean, 
                                      ymax=group_mean),
                width=0.4, color = "black")
  scale_y_continuous(name = "Richness (Observed ESVs)",
                     limits = c(0, 40),
                     breaks = c(0, 10, 20, 30, 40))
richR
```

### Gala
Subset sample data for gala only
```{r include=FALSE}
gala_data <- read_delim(file = "Sample_metadata_file_16S.csv", delim = ",", 
                       col_names = TRUE) %>%
  filter(Cultivar == "Gala")
```

Calculate alpha diversity metrics, and format data for plotting. 
```{r}
richness_dfg <- psq %>%
  # subset phyloseq object for cultivar 
  subset_samples(Cultivar == "Gala") %>% 
  # calculate richness (observed OTUs)
  estimate_richness(., split = TRUE,  measures = c("Observed", "Shannon")) %>% 
  # make sample_id column before join 
  rownames_to_column(var = "Sample_ID") %>% 
  # join with sample data imported above
  inner_join(gala_data, by = "Sample_ID") %>%  
  rename(Richness = Observed) %>%
  # calculate average richness per group 
  group_by(Tissue, Tree, Timepoint, Cultivar, Treatment) %>%
  summarise(Avg_Richness = round(mean(Richness), digits = 0),
            Avg_Shannon = round(mean(Shannon), digits = 2)) %>%
  mutate(Timepoint = recode(Timepoint,
                             "1"= "Day 0",
                             "2" = "Day 14",
                             "3" = "Day 28",
                             "4" = "Day 42",
                             "5" = "Day 56"),
         Timepoint = factor(Timepoint, ordered = TRUE,
                            levels = c("Day 0", "Day 14", "Day 28",
                                       "Day 42", "Day 56", "Rotten"))) 
```
Generate gala richness plot.
```{r include=FALSE}
richg1 <- psq %>%
  # subset phyloseq object for cultivar 
  subset_samples(Cultivar == "Gala") %>% 
  # make the plot 
  plot_richness(., "Timepoint", measures = c("Observed")) +
  facet_grid(~Tissue) + xlab(NULL)
richg1
```

Get data for error bars 
```{r}
# manually create temp data frame for rotten SDs
tmp <- data.frame(Tissue = c("Pulp", "Skin"),
                  Timepoint = c("Rotten", "Rotten"),
                  Treatment = c("Untreated", "Untreated"),
                  group_mean = c(16.000000, 27.000000),
                  group_sd = c(3.05505, 11.54701),
                  stringsAsFactors = FALSE) %>%
    mutate(Timepoint = recode(Timepoint,
                             "1"= "Day 0",
                             "2" = "Day 14",
                             "3" = "Day 28",
                             "4" = "Day 42",
                             "5" = "Day 56"),
         Timepoint = factor(Timepoint, ordered = TRUE,
                            levels = c("Day 0", "Day 14", "Day 28",
                                       "Day 42", "Day 56", "Rotten")))

# lines for error bars
line_posits <- richness_dfg %>% 
  filter(Cultivar == "Gala",
         Treatment == "Untreated") %>%
  group_by(Tissue, Timepoint, Treatment) %>%
  summarise(group_mean = mean(Avg_Richness, na.rm = TRUE), 
            group_sd = sd(Avg_Richness, na.rm = TRUE)) %>%
  ungroup() %>%
  # add SD values from temp dataframe
  drop_na() %>%
  rbind(tmp)
```

ANOVA and TukeyHSDfor pulp and skin of Gala, get letter annotations for plot
```{r}
AOVg1 <- filter(richness_dfg, Tissue == "Pulp") %>%
  aov(data = ., Avg_Richness~Timepoint) 
summary(AOVg1) #p=0.000643
tuk1 <- tidy(TukeyHSD(AOVg1)) %>%
  mutate(adj.p.value = round(adj.p.value, digits = 4))
letters1 <- as.matrix(HSD.test(AOVg1,"Timepoint", group=TRUE, alpha = 0.05)$groups) %>%
  as.data.frame() %>%
  rownames_to_column(var="Timepoint") %>%
  mutate(Avg_Richness = round(as.numeric(as.character(Avg_Richness)), digits=0),
         groups = as.character(groups))  %>%
  add_column(Tissue = "Pulp")

AOVg2 <- filter(richness_dfg, Tissue == "Skin") %>%
  aov(data = ., Avg_Richness~Timepoint)
summary(AOVg2) #p=0.00525
tuk2 <- tidy(TukeyHSD(AOVg2)) %>%
  mutate(adj.p.value = round(adj.p.value, digits = 4))
letters2 <- as.matrix(HSD.test(AOVg2,"Timepoint", group=TRUE, alpha = 0.05)$groups) %>%
  as.data.frame() %>%
  rownames_to_column(var="Timepoint") %>%
  mutate(Avg_Richness = round(as.numeric(as.character(Avg_Richness)), digits=0),
         groups = as.character(groups)) %>%
  add_column(Tissue = "Skin")
```
Combine letter data frames with line position data frame 
```{r}
lettersG <- rbind(letters1, letters2)
line_posits2 <- inner_join(line_posits, lettersG, by = c("Tissue", "Timepoint"))
```

Make it pretty. 
```{r}
richg1$layers <- richg1$layers[-1] #drop geom_point 

richG1 <- richg1 +   
  # add individual points in color
  geom_jitter(aes(color = Timepoint), width = 0.3) +  
  # plot standard deviation
  geom_errorbar(data=line_posits2, aes(x=Timepoint, 
                                      y=group_mean,  
                                      ymin=group_mean-group_sd, 
                                      ymax=group_mean+group_sd), 
                width=0.1, color = "black") +
  # plot mean line 
  geom_errorbar(data=line_posits2, aes(x=Timepoint, 
                                      y=group_mean,  
                                      ymin=group_mean, 
                                      ymax=group_mean),
                width=0.4, color = "black") +
  # add letter annotations
  geom_text(data=line_posits2, inherit.aes = FALSE,
            aes(x=Timepoint, y=49, label=groups)) +
  # adjust y-axis
  scale_y_continuous(name = "Richness",
                     limits = c(0, 50),
                     breaks = c(0, 10, 20, 30, 40, 50)) + 
  # change colors 
  scale_color_manual(name = "Timepoint",
                     breaks = c("Day 0", "Day 14", "Day 28", 
                                "Day 42", "Rotten"),
                     values = c("#ED1E24", "#9ACA3C", "#40B9EB",
                                "#426FB6", "#D771AD")) +
  theme(legend.position = "none") + 
  ggtitle("Gala") 

richG1
```

Generate gala richness plot *without* rotten samples
```{r include=FALSE}
richg2 <- psq %>%
  # subset phyloseq object for cultivar
  subset_samples(Timepoint != "Rotten") %>%
  subset_samples(Cultivar == "Gala") %>% 
  # make the plot 
  plot_richness(., "Timepoint", measures = c("Observed")) +
  facet_grid(~Tissue) + xlab(NULL)
richg2
```

Make it pretty. 
```{r}
richg2$layers <- richg2$layers[-1] #drop geom_point 

richG2 <- richg2 +   
  geom_violin(aes(color = Timepoint)) + #add violin in color
  geom_jitter(aes(color = Timepoint)) +  #add individual points in color
  theme(legend.position = "none") +
  ggtitle("Gala") + 
  scale_y_continuous(name = "Richness (Observed ESVs)",
                     limits = c(0, 50),
                     breaks = c(0, 10, 20, 30, 40, 50))
richG2
```

### Jonathan
Subset data by cultivar
```{r}
jon_data <- read_delim(file = "Sample_metadata_file_16S.csv", delim = ",", 
                       col_names = TRUE) %>%
  # subset sample metadata for groups of interest 
  filter(Cultivar == "Jonathan",
         Treatment == "Untreated")
```

Calculate richness metrics
```{r}
richness_dfj <- psq %>%
  # subset phyloseq object for cultivar
  subset_samples(Cultivar == "Jonathan") %>%
  subset_samples(Treatment == "Untreated") %>% 
  # calculate richness (observed OTUs)
  estimate_richness(., split = TRUE,  measures = c("Observed", "Shannon")) %>% 
  # make sample_id column before join 
  rownames_to_column(var = "Sample_ID") %>% 
  # join with sample data imported above
  inner_join(jon_data, by = "Sample_ID") %>%  
  rename(Richness = Observed) %>%
  # calculate average richness per group 
  group_by(Tissue, Tree, Timepoint, Cultivar, Treatment) %>%
  summarise(Avg_Richness = round(mean(Richness), digits = 0),
            Avg_Shannon = round(mean(Shannon), digits = 2)) %>%
  mutate(Timepoint = recode(Timepoint,
                             "1"= "Day 0",
                             "2" = "Day 14",
                             "3" = "Day 28",
                             "4" = "Day 42",
                             "5" = "Day 56"),
         Timepoint = factor(Timepoint, ordered = TRUE,
                            levels = c("Day 0", "Day 14", "Day 28",
                                       "Day 42", "Day 56", "Rotten"))) 
```

Generate jonathan plot.
```{r include=FALSE}
# Plot richness
richj1 <- psq %>%
  # filter phyloseq object for cultivar
  subset_samples(Cultivar == "Jonathan") %>%
  subset_samples(Treatment == "Untreated") %>% 
  # make the plot 
  plot_richness(., "Timepoint", measures = c("Observed")) +
  facet_grid(~Tissue) + xlab(NULL)
richj1
```

Get values for error bars 
```{r}
# manually make SD for rotten samples with temp data frame
tmp <- data.frame(Tissue = c("Pulp", "Skin"),
                  Timepoint =c("Rotten", "Rotten"),
                  Treatment = c("Untreated", "Untreated"),
                  group_mean = c(19.00000, 20.00000),
                  group_sd = c(6.377042, 3.201562),
                  stringsAsFactors = FALSE) %>%
  mutate(Timepoint = recode(Timepoint,
                            "1"= "Day 0",
                            "2" = "Day 14",
                            "3" = "Day 28",
                            "4" = "Day 42",
                            "5" = "Day 56"),
         Timepoint = factor(Timepoint, ordered = TRUE,
                            levels = c("Day 0", "Day 14", "Day 28",
                                       "Day 42", "Day 56", "Rotten"))) 
# lines for error bars
line_posits <- richness_dfj %>% 
  filter(Cultivar == "Jonathan",
         Treatment == "Untreated") %>%
  group_by(Tissue, Timepoint, Treatment) %>%
  summarise(group_mean = mean(Avg_Richness, na.rm = TRUE), 
            group_sd = sd(Avg_Richness, na.rm = TRUE)) %>%
  ungroup() %>% 
  #add missing SD values from temp dataframe
  drop_na() %>%
  rbind(tmp)
```

ANOVA for pulp and skin of Jonathan 
```{r}
AOVj1 <- filter(richness_dfj, Tissue == "Pulp") %>%
  aov(data = ., Avg_Richness~Timepoint) 
summary(AOVj1) #p=0.189
tukj1 <- tidy(TukeyHSD(AOVj1)) %>%
  mutate(adj.p.value = round(adj.p.value, digits = 4))
lettersj1 <- as.matrix(HSD.test(AOVj1,"Timepoint", group=TRUE, alpha = 0.05)$groups) %>%
  as.data.frame() %>%
  rownames_to_column(var="Timepoint") %>%
  mutate(Avg_Richness = round(as.numeric(as.character(Avg_Richness)), digits=0),
         groups = as.character(groups)) %>%
  add_column(Tissue = "Pulp")

AOVj2 <- filter(richness_dfj, Tissue == "Skin") %>%
  aov(data = ., Avg_Richness~Timepoint) 
summary(AOVj2) #p=0.761
tukj2 <- tidy(TukeyHSD(AOVj2)) %>%
  mutate(adj.p.value = round(adj.p.value, digits = 4))
lettersj2 <- as.matrix(HSD.test(AOVj2,"Timepoint", group=TRUE, alpha = 0.05)$groups) %>%
  as.data.frame() %>%
  rownames_to_column(var="Timepoint") %>%
  mutate(Avg_Richness = round(as.numeric(as.character(Avg_Richness)), digits=0),
         groups = as.character(groups)) %>%
  add_column(Tissue = "Skin")
```

Combine letter data frames with line position data frame 
```{r}
lettersJ <- rbind(lettersj1, lettersj2)
line_posits2 <- inner_join(line_posits, lettersJ, by = c("Tissue", "Timepoint"))
```

Make it pretty. 
```{r}
richj1$layers <- richj1$layers[-1] #drop geom_point 

richJ1 <- richj1 +   
  geom_jitter(aes(color = Timepoint), width = 0.3) +  #add individual points in color 
  # plot standard deviation
  geom_errorbar(data=line_posits, aes(x=Timepoint, 
                                      y=group_mean,  
                                      ymin=group_mean-group_sd, 
                                      ymax=group_mean+group_sd), 
                width=0.1, color = "black") +
  # plot mean line 
  geom_errorbar(data=line_posits, aes(x=Timepoint, 
                                      y=group_mean,  
                                      ymin=group_mean, 
                                      ymax=group_mean),
                width=0.4, color = "black") +
  # add letter annotations
  geom_text(data=line_posits2, inherit.aes = TRUE, 
            aes(x = Timepoint, y = 49, label = groups)) + 
  scale_y_continuous(name = "Richness",
                     limits = c(0, 50),
                     breaks = c(0, 10, 20, 30, 40, 50)) +
  # change colors 
  scale_color_manual(name = "Timepoint",
                     breaks = c("Day 0", "Day 14", "Day 28", 
                                "Day 42", "Rotten"),
                     values = c("#ED1E24", "#9ACA3C", "#40B9EB",
                                "#426FB6", "#D771AD")) +
  theme(legend.position = "none") + 
  ggtitle("Jonathan") 

richJ1
```

Generate jonathan plot *without* rotten timepoint
```{r}
# Plot richness
richj2 <- psq %>%
  # filter phyloseq object for cultivar
  subset_samples(Timepoint != "Rotten") %>%
  subset_samples(Cultivar == "Jonathan") %>%
  subset_samples(Treatment == "Untreated") %>%
  # make the plot 
  plot_richness(., "Timepoint", measures = c("Observed")) +
  facet_grid(~Tissue) + xlab(NULL)
richj2
```

Make it pretty. 
```{r}
richj2$layers <- richj2$layers[-1] #drop geom_point 

richJ2 <- richj2 +   
  geom_violin(aes(color = Timepoint)) + #add violin in color
  geom_jitter(aes(color = Timepoint)) +  #add individual points in color
  theme(legend.position = "none") + 
  ggtitle("Jonathan") +
  scale_y_continuous(name = "Richness (Observed ESVs)",
                     limits = c(0, 50),
                     breaks = c(0, 10, 20, 30, 40, 50))
richJ2
```

### Combine cultivars
With rotten timepoint
```{r}
# horizontal
richGJ1 <- plot_grid(richG1, richJ1,
                     nrow = 1, ncol = 2)
richGJ1
#vertical
richGJ3 <- plot_grid(richG1, richJ1,
                     nrow = 2, ncol = 1)
richGJ3
```

With out rotten timepoint
```{r}
richGJ2 <- plot_grid(richG2, richJ2,
                     nrow = 1, ncol = 2)
richGJ2
```

### Statistics
```{r}

```


### Save plots
```{r}
# shelf life, untreated, no rotten 
save_plot(plot = richU, 
          base_height = 7, base_width = 5,
          filename = "Figures/Richness_untreated_timecourse.pdf") 
# chitosan treatments, jonathan only
save_plot(richCj, 
          base_height = 7, base_width = 5,
          filename = "Figures/Richness_chitosan_jonathan.pdf")
# rotten samples 
save_plot(richR, 
          base_height = 7, base_width = 5,
          filename = "Figures/Richness_rotten.pdf")
# cultivars untreated over time
save_plot(richGJ1, ncol = 2, nrow = 1,
          base_height = 5, base_width = 7,
          filename = "Figures/Richness_cultivars_Apr8.pdf")
save_plot(richGJ2, ncol = 2, nrow = 1,
          filename = "Figures/Richness_cultivars_v2.pdf")


```

# Shannon
The code in this section is essentially the same as everything in the Richness section, but it is plotting the Shannon index instead of Observed OTUs. 

### Gala
Generate gala plot.
```{r}
shang1 <- psq %>%
  # subset phyloseq object for cultivar 
  subset_samples(Cultivar == "Gala") %>% 
  subset_samples(Treatment == "Untreated") %>%
  # make the plot 
  plot_richness(., "Timepoint", measures = c("Shannon")) +
  facet_grid(~Tissue) + xlab(NULL)
shang1
```

```{r}
# manually create temp data frame for rotten SDs
tmp <- data.frame(Tissue = c("Pulp", "Skin"),
                  Timepoint = c("Rotten", "Rotten"),
                  Treatment = c("Untreated", "Untreated"),
                  group_mean = c(0.1342863, 0.1994544),
                  group_sd = c(0.03418375, 0.04885647),
                  stringsAsFactors = FALSE) %>%
    mutate(Timepoint = recode(Timepoint,
                             "1"= "Day 0",
                             "2" = "Day 14",
                             "3" = "Day 28",
                             "4" = "Day 42",
                             "5" = "Day 56"),
         Timepoint = factor(Timepoint, ordered = TRUE,
                            levels = c("Day 0", "Day 14", "Day 28",
                                       "Day 42", "Day 56", "Rotten")))
```

```{r}
# lines for error bars
line_posits <- richness_dfg %>% 
  filter(Cultivar == "Gala",
         Treatment == "Untreated") %>%
  group_by(Tissue, Timepoint, Treatment) %>%
  summarise(group_mean = mean(Avg_Shannon, na.rm = TRUE), 
            group_sd = sd(Avg_Shannon, na.rm = TRUE)) %>%
  ungroup() %>%
  # add SD values from temp dataframe
  drop_na() %>%
  rbind(tmp)
```
Make it pretty. 
```{r}
shang1$layers <- shang1$layers[-1] #drop geom_point 

shanG1 <- shang1 +   
  # add individual points in color
  geom_jitter(aes(color = Timepoint), width = 0.3) +  
  # plot standard deviation
  geom_errorbar(data=line_posits, aes(x=Timepoint, 
                                      y=group_mean,  
                                      ymin=group_mean-group_sd, 
                                      ymax=group_mean+group_sd), 
                width=0.1, color = "black") +
  # plot mean line 
  geom_errorbar(data=line_posits, aes(x=Timepoint, 
                                      y=group_mean,  
                                      ymin=group_mean, 
                                      ymax=group_mean),
                width=0.4, color = "black") +
  # # adjust y-axis
  # scale_y_continuous(name = "Shannon Index",
  #                    limits = c(-0.3, 1.6),
  #                    breaks = c(0, 0.25, 0.50, 0.75, 1.0, 1.25, 1.5)) +
  # change colors 
  scale_color_manual(name = "Timepoint",
                     breaks = c("Day 0", "Day 14", "Day 28", 
                                "Day 42", "Rotten"),
                     values = c("#ED1E24", "#9ACA3C", "#40B9EB",
                                "#426FB6", "#D771AD")) +
  theme(legend.position = "none") + 
  ggtitle("Gala") + ylab("Shannon Index")

shanG1
```

### Jonathan
Generate plot.
```{r include=FALSE}
shanj1 <- psq %>%
  # subset phyloseq object for cultivar 
  subset_samples(Cultivar == "Jonathan") %>% 
  subset_samples(Treatment == "Untreated") %>%
  # make the plot 
  plot_richness(., "Timepoint", measures = c("Shannon")) +
  facet_grid(~Tissue) + xlab(NULL)
shanj1
```

```{r}
# manually create temp data frame for rotten SDs
tmp <- data.frame(Tissue = c("Pulp", "Skin"),
                  Timepoint = c("Rotten", "Rotten"),
                  Treatment = c("Untreated", "Untreated"),
                  group_mean = c(0.1942355, 0.1254453),
                  group_sd = c(0.07665664, 0.04009325),
                  stringsAsFactors = FALSE) %>%
    mutate(Timepoint = recode(Timepoint,
                             "1"= "Day 0",
                             "2" = "Day 14",
                             "3" = "Day 28",
                             "4" = "Day 42",
                             "5" = "Day 56"),
         Timepoint = factor(Timepoint, ordered = TRUE,
                            levels = c("Day 0", "Day 14", "Day 28",
                                       "Day 42", "Day 56", "Rotten")))
```

```{r}
# lines for error bars
line_posits <- richness_dfj %>% 
  filter(Cultivar == "Jonathan",
         Treatment == "Untreated") %>%
  group_by(Tissue, Timepoint, Treatment) %>%
  summarise(group_mean = mean(Avg_Shannon, na.rm = TRUE), 
            group_sd = sd(Avg_Shannon, na.rm = TRUE)) %>%
  ungroup() %>%
  # add SD values from temp dataframe
  drop_na() %>%
  rbind(tmp)
```
Make it pretty. 
```{r}
shanj1$layers <- shanj1$layers[-1] #drop geom_point 

shanJ1 <- shanj1 +   
  # add individual points in color
  geom_jitter(aes(color = Timepoint), width = 0.3) +  
  # plot standard deviation
  geom_errorbar(data=line_posits, aes(x=Timepoint, 
                                      y=group_mean,  
                                      ymin=group_mean-group_sd, 
                                      ymax=group_mean+group_sd), 
                width=0.1, color = "black") +
  # plot mean line 
  geom_errorbar(data=line_posits, aes(x=Timepoint, 
                                      y=group_mean,  
                                      ymin=group_mean, 
                                      ymax=group_mean),
                width=0.4, color = "black") +
  # # adjust y-axis
  # scale_y_continuous(name = "Shannon Index",
  #                    limits = c(-0.1, 1.6),
  #                    breaks = c(0, 0.25, 0.50, 0.75, 1.0, 1.25, 1.5)) +
  # change colors 
  scale_color_manual(name = "Timepoint",
                     breaks = c("Day 0", "Day 14", "Day 28", 
                                "Day 42", "Rotten"),
                     values = c("#ED1E24", "#9ACA3C", "#40B9EB",
                                "#426FB6", "#D771AD")) +
  theme(legend.position = "none") + 
  ggtitle("Jonathan") + ylab("Shannon Index")

shanJ1
```

### Save plots
```{r}
save_plot(shanG1, ncol = 1, nrow = 1,
          filename = "Figures/Shannon_gala.pdf")

save_plot(shanJ1, ncol = 1, nrow = 1,
          filename = "Figures/Shannon_jonathan.pdf")
```

# Taxa
Are the abundances of any of these groups different between timepoints?
Firmicutes, Bacteroidetes, Proteobacteria (Alpha, Beta, Gamma), Actinobacteria
```{r}

```

# Export
```{r}

```

-----
end
