---
title: "Rot Microbiome 3"
author: "Kristi Gdanetz MacCready"
date: "4/5/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = "~/Documents/Sundin_Lab/Rot_microbiome/") #set working directory 
```

# Load packages
load R packages and custom functions 
```{r}
library(vegan)
library(tidyverse)
library(readxl)
library(agricolae)
library(phyloseq)
library(venn)
library(VennDiagram)
library(ggvenn)
library(cowplot)
theme_set(theme_cowplot())
set.seed(7)
source("~/Documents/GitHub/DenefLab-MicrobeMiseq/R/miseqR.R") #use custom functions 
```

# Import data

### Phyloseq object
Import object created in AlphaDiversity_analysis.Rmd 
```{r}
psq <- readRDS(file = "psq_object.RDS")
psq
```

### Taxa
taxonomy table
```{r}
tax <- read_delim(file = "RotOTUs.taxonomy_edited.csv", delim = ",", 
                       col_names = TRUE) %>%
  select(-"Size", -"D", -"P", -"C", -"O", -"F", -"G") 
saveRDS(object = tax, file = "tax.RDS")
```

### Metadata 
sample metadata
```{r}
metadata <- read_delim(file = "Sample_metadata_file_16S.csv", delim = ",", 
                       col_names = TRUE) %>%
  mutate(Timepoint = recode(Timepoint,
                             "1"= "Day 0",
                             "2" = "Day 14",
                             "3" = "Day 28",
                             "4" = "Day 42",
                             "5" = "Day 56"))
```

# Format shared tables
### Phylum 
Manually format shared (Otu) table. Summarise OTU counts by phylum
```{r}
shared_phylum <- read_delim(file = "RotOTUs.shared", delim = "\t", 
                       col_names = TRUE) %>% #import shared data
  select(-label, -numOtus) %>%
  # format to join with tax data
  column_to_rownames(var = "Group") %>%
  base::t() %>%
  as.data.frame() %>%
  rownames_to_column(var = "OTU") %>%
  # join and reorder columns
  full_join(tax, by = "OTU") %>%
  select(OTU, Domain, Phylum, Class, Order, Family, Genus, everything()) %>%
  # group by phyla, sum OTUs in same taxonomic group
  group_by(Phylum) %>%
  summarise_if(is.numeric, sum) %>%
  ungroup() %>% 
  # transpose table again
  column_to_rownames(var = "Phylum") %>%
  base::t() %>%
  as.data.frame() %>%
  rownames_to_column(var = "Sample_ID") %>%
  # add sample data, reorder columns 
  inner_join(metadata, by = "Sample_ID") %>%
  select(Sample_ID, Cultivar, Treatment, Tissue, Timepoint, Tree, Replicate, everything()) %>%
  arrange(Cultivar, Treatment, Timepoint) %>%
  filter(Cultivar != "Cherry")
dim(shared_phylum) #346 x 28
saveRDS(object = shared_phylum, file = "shared_phylum.RDS")
```

### Class
```{r}
shared_class <- read_delim(file = "RotOTUs.shared", delim = "\t", 
                       col_names = TRUE) %>% #import shared data
  select(-label, -numOtus) %>%
  # format to join with tax data
  column_to_rownames(var = "Group") %>%
  base::t() %>%
  as.data.frame() %>%
  rownames_to_column(var = "OTU") %>%
  # join and reorder columns
  full_join(tax, by = "OTU") %>%
  select(OTU, Domain, Phylum, Class, Order, Family, Genus, everything()) %>%
  # group by phyla, sum OTUs in same taxonomic group
  group_by(Class) %>%
  summarise_if(is.numeric, sum) %>%
  ungroup() %>% 
  # transpose table again
  column_to_rownames(var = "Class") %>%
  base::t() %>%
  as.data.frame() %>%
  rownames_to_column(var = "Sample_ID") %>%
  # add sample data, reorder columns 
  inner_join(metadata, by = "Sample_ID") %>%
  select(Sample_ID, Cultivar, Treatment, Tissue, Timepoint, Tree, Replicate, everything()) %>%
  arrange(Cultivar, Treatment, Timepoint) %>%
  filter(Cultivar != "Cherry")
dim(shared_class) #346 x 66
```

### Order
```{r}
shared_order <- read_delim(file = "RotOTUs.shared", delim = "\t", 
                       col_names = TRUE) %>% #import shared data
  select(-label, -numOtus) %>%
  # format to join with tax data
  column_to_rownames(var = "Group") %>%
  base::t() %>%
  as.data.frame() %>%
  rownames_to_column(var = "OTU") %>%
  # join and reorder columns
  full_join(tax, by = "OTU") %>%
  select(OTU, Domain, Phylum, Class, Order, Family, Genus, everything()) %>%
  # group by phyla, sum OTUs in same taxonomic group
  group_by(Order) %>%
  summarise_if(is.numeric, sum) %>%
  ungroup() %>% 
  # transpose table again
  column_to_rownames(var = "Order") %>%
  base::t() %>%
  as.data.frame() %>%
  rownames_to_column(var = "Sample_ID") %>%
  # add sample data, reorder columns 
  inner_join(metadata, by = "Sample_ID") %>%
  select(Sample_ID, Cultivar, Treatment, Tissue, Timepoint, Tree, Replicate, everything()) %>%
  arrange(Cultivar, Treatment, Timepoint) %>%
  filter(Cultivar != "Cherry")
dim(shared_order) #346 x 109
```

### Family
```{r}
shared_family <- read_delim(file = "RotOTUs.shared", delim = "\t", 
                       col_names = TRUE) %>% #import shared data
  select(-label, -numOtus) %>%
  # format to join with tax data
  column_to_rownames(var = "Group") %>%
  base::t() %>%
  as.data.frame() %>%
  rownames_to_column(var = "OTU") %>%
  # join and reorder columns
  full_join(tax, by = "OTU") %>%
  select(OTU, Domain, Phylum, Class, Order, Family, Genus, everything()) %>%
  # group by phyla, sum OTUs in same taxonomic group
  group_by(Family) %>%
  summarise_if(is.numeric, sum) %>%
  ungroup() %>% 
  # transpose table again
  column_to_rownames(var = "Family") %>%
  base::t() %>%
  as.data.frame() %>%
  rownames_to_column(var = "Sample_ID") %>%
  # add sample data, reorder columns 
  inner_join(metadata, by = "Sample_ID") %>%
  select(Sample_ID, Cultivar, Treatment, Tissue, Timepoint, Tree, Replicate, everything()) %>%
  arrange(Cultivar, Treatment, Timepoint) %>%
  filter(Cultivar != "Cherry")
dim(shared_family) #346 x 196
```

### Genus
```{r}
shared_genus <- read_delim(file = "RotOTUs.shared", delim = "\t", 
                       col_names = TRUE) %>% #import shared data
  select(-label, -numOtus) %>%
  # format to join with tax data
  column_to_rownames(var = "Group") %>%
  base::t() %>%
  as.data.frame() %>%
  rownames_to_column(var = "OTU") %>%
  # join and reorder columns
  full_join(tax, by = "OTU") %>%
  select(OTU, Domain, Phylum, Class, Order, Family, Genus, everything()) %>%
  # group by phyla, sum OTUs in same taxonomic group
  group_by(Genus) %>%
  summarise_if(is.numeric, sum) %>%
  ungroup() %>% 
  # transpose table again
  column_to_rownames(var = "Genus") %>%
  base::t() %>%
  as.data.frame() %>%
  rownames_to_column(var = "Sample_ID") %>%
  # add sample data, reorder columns 
  inner_join(metadata, by = "Sample_ID") %>%
  select(Sample_ID, Cultivar, Treatment, Tissue, Timepoint, Tree, Replicate, everything()) %>%
  arrange(Cultivar, Treatment, Timepoint) %>%
  filter(Cultivar != "Cherry")
dim(shared_genus) #346 x 336
```

### OTUs
```{r}
shared_meta <- read_delim(file = "RotOTUs.shared", delim = "\t", 
                       col_names = TRUE) %>% #import shared data
  select(-label, -numOtus) %>%
  rename(Sample_ID = Group) %>% 
  # add sample data, reorder columns 
  inner_join(metadata, by = "Sample_ID") %>%
  select(Sample_ID, Cultivar, Treatment, Tissue, Timepoint, Tree, Replicate, starts_with("Otu")) %>%
  arrange(Cultivar, Treatment, Timepoint) %>%
  filter(Cultivar != "Cherry")
dim(shared_meta) #346 x 337
```

# Most abundant genera
```{r}
# make temp dataframe to sum OTUs in each genus x sample
tmp1 <- shared_genus %>%
  #remove chitosan treated samples
  filter(Treatment == "Untreated") %>% 
  # sum replicates samples 
  select(-Replicate) %>%
  group_by(Cultivar, Tissue, Timepoint) %>%
  summarize_if(is.numeric, sum) %>%
  ungroup() %>% 
  # make new sample names 
  mutate(Sample = paste(Cultivar, Tissue, Timepoint, sep = "-")) %>%
  select(Sample, everything(), -Cultivar, -Tissue, -Timepoint) %>%
  column_to_rownames(var = "Sample") %>%
  # sum read counts for each OTU
  colSums() %>%
  as.data.frame() %>%
  rename(OtuSum = ".") %>%
  arrange(OtuSum) %>%
  rownames_to_column(var = "Genus") 

# join back with sample data, remove <10 reads, chloroplasts 
tmp2 <- shared_genus %>%
  filter(Treatment == "Untreated") %>% 
  select(-Replicate) %>%
  group_by(Cultivar, Tissue, Timepoint) %>%
  summarize_if(is.numeric, sum) %>%
  ungroup() %>% 
  mutate(Sample = paste(Cultivar, Tissue, Timepoint, sep = "-")) %>%
  select(Sample, everything(), -Cultivar, -Tissue, -Timepoint) %>%
  column_to_rownames(var = "Sample") %>%
  t() %>%
  as.data.frame() %>%
  rownames_to_column(var = "Genus") %>%
  full_join(tmp1, by = "Genus") %>%
  filter(OtuSum >= 10,
         Genus != "Chloroplast_unclassified",
         Genus != "Cyanobacteria/Chloroplast_unclassified",
         Genus != "Streptophyta_unclassified") 

write_delim(tmp2, file = "tmp2.csv", delim = ",", col_names = TRUE)
```

```{r}

# remove chloroplasts

# merge at genus level

# make long data frame 

# calculate relative abundance

tmp3 <- psq %>%
  #prune_taxa(taxa_sums(.) >= 10) %>%
  subset_taxa(Phylum != "Cyanobacteria/Chloroplast") %>%
  subset_samples(Treatment == "Untreated") %>%
  tax_glom("Genus") %>%
  #sample_sums()
  transform_sample_counts(function(x) { (x / sum(x) ) * 100} ) %>%
  psmelt() 
```

# Venn Diagrams
```{r}
# OTU table subsetted for untreated samples
df1 <- shared_meta %>%
  filter(Treatment == "Untreated") %>%
  select(-Replicate) %>%
  group_by(Cultivar, Treatment, Tissue, Timepoint, Tree) %>%
  summarise_if(is.numeric, mean) %>%
  mutate_if(is.numeric, round, digits = 0) %>%
  ungroup() %>%
  mutate(tmp = paste(Cultivar, Treatment, Tissue, Timepoint, Tree, sep = "-")) %>%
  select(tmp, starts_with("Otu"), 
         -"Cultivar", -"Treatment", -"Tissue", -"Timepoint", -"Tree") %>%
  column_to_rownames(var = "tmp") 

# use copy of subsetted otu table to calculate total reads per otu
df2 <- df1 %>%
  colSums() %>%
  as.data.frame() %>%
  rename(OtuSum = ".") %>%
  base::t() %>%
  as.data.frame()
  
# combine otu table with sums, remove otus with less than 10 reads 
df3 <- rbind(df1, df2) %>%
  base::t() %>%
  as.data.frame() %>% #330x53
  filter(OtuSum > 10) %>% #48x53
  base::t() %>%
  as.data.frame() %>%
  rownames_to_column(var = "Sample_ID") %>%
  select(Sample_ID, starts_with("Otu")) %>%
  filter(Sample_ID != "OtuSum")

# convert to binary
df4 <- df3
df4[-1] <- sapply(df4[-1], function(x) { as.numeric(x > 0) })
```

### Both cultivars
Venn diagram between cultivars, over time, both tissues combined.
```{r}
dfB <- df4 %>%
  # get categories back
  separate(col = "Sample_ID",
           into = c("Cultivar", "Treatment", "Tissue", "Timepoint", "Tree"),
           sep = "-",
           remove = FALSE) %>% 
  # subset by cultivar
  #filter(Cultivar == "Jonathan") %>%
  # format variables by cultivar and time
  group_by(Timepoint, Cultivar) %>%
  summarise_if(is.numeric, sum) %>%
  ungroup() %>%
  mutate(Sample_ID = paste(Cultivar, Timepoint, sep = "-")) %>%
  select(Sample_ID, starts_with("Otu"), -Cultivar, -Timepoint)

dfB2 <- dfB %>%
  # transpose for venn format
  column_to_rownames("Sample_ID") %>%
  t() %>% as.data.frame() %>%
  # remove OTUs not found in any samples
  #rowSums() 
  rownames_to_column(var = "Otu") %>%
  # filter() %>% 
  # convert binary to logical
  mutate_if(is.numeric, as.logical)

v13 <- ggvenn(dfB2, c("Gala-Day 0", "Jonathan-Day 0"),
      fill_color = c("#ED1E24", "#71C054"),
      stroke_size = 0.5, digits = 0,
      set_name_size = 4) + ggtitle("Day 0")
v14 <- ggvenn(dfB2, c("Gala-Day 14", "Jonathan-Day 14"),
      fill_color = c("#ED1E24", "#71C054"),
      stroke_size = 0.5, digits = 0,
      set_name_size = 4) + ggtitle("Day 14")
v15 <- ggvenn(dfB2, c("Gala-Day 28", "Jonathan-Day 28"),
      fill_color = c("#ED1E24", "#71C054"),
      stroke_size = 0.5, digits = 0,
      set_name_size = 4) + ggtitle("Day 28")
v16 <- ggvenn(dfB2, c("Gala-Day 42", "Jonathan-Day 42"),
      fill_color = c("#ED1E24", "#71C054"),
      stroke_size = 0.5, digits = 0,
      set_name_size = 4) +  ggtitle("Day 42")
v17 <- ggvenn(dfB2, c("Gala-Rotten", "Jonathan-Rotten"),
      fill_color = c("#ED1E24", "#71C054"),
      stroke_size = 0.5, digits = 0,
      set_name_size = 4) + ggtitle("Rotten")

v18 <- plot_grid(v13, v14, v15, v16, v17,
                nrow = 5, ncol = 1)
```

### Jonathan
Venn diagram showing shared OTUs between skin and pulp over time
```{r}
# Venn dataframe for Jonathan
dfj <- df4 %>%
  # get categories back
  separate(col = "Sample_ID",
           into = c("Cultivar", "Treatment", "Tissue", "Timepoint", "Tree"),
           sep = "-",
           remove = FALSE) %>% 
  # subset by cultivar
  filter(Cultivar == "Jonathan") %>%
  # format variables by tissue and time
  group_by(Timepoint, Tissue) %>%
  summarise_if(is.numeric, sum) %>%
  ungroup() %>%
  mutate(Sample_ID = paste(Tissue, Timepoint, sep = "-")) %>%
  select(Sample_ID, starts_with("Otu"), -Tissue, -Timepoint)

dfj3 <- dfj %>%
  # transpose for venn format
  column_to_rownames("Sample_ID") %>%
  t() %>% as.data.frame() %>%
  # remove OTUs not found in any samples
  #rowSums() #Otu035, Otu068, Otu072	= 0
  rownames_to_column(var = "Otu") %>%
  filter(Otu != "Otu035",
         Otu != "Otu068",
         Otu != "Otu072") %>% 
  # convert binary to logical
  mutate_if(is.numeric, as.logical)

v1 <- ggvenn(dfj3, c("Pulp-Day 0", "Skin-Day 0"),
      fill_color = c("#C5E2B5", "#71C054"),
      stroke_size = 0.5, 
      digits = 0,
      set_name_size = 4) + ggtitle("Jonathan Day 0")
v2 <- ggvenn(dfj3, c("Pulp-Day 14", "Skin-Day 14"),
      fill_color = c("#C5E2B5", "#71C054"),
      stroke_size = 0.5, 
      digits = 0,
      set_name_size = 4) + ggtitle("Jonathan Day 14")
v3 <- ggvenn(dfj3, c("Pulp-Day 28", "Skin-Day 28"),
      fill_color = c("#C5E2B5", "#71C054"),
      stroke_size = 0.5, 
      digits = 0,
      set_name_size = 4) + ggtitle("Jonathan Day 28")
v4 <- ggvenn(dfj3, c("Pulp-Day 42", "Skin-Day 42"),
      fill_color = c("#C5E2B5", "#71C054"),
      stroke_size = 0.5, 
      digits = 0,
      set_name_size = 4) +  ggtitle("Jonathan Day 42")
v5 <- ggvenn(dfj3, c("Pulp-Rotten", "Skin-Rotten"),
      fill_color = c("#C5E2B5", "#71C054"),
      stroke_size = 0.5, 
      digits = 0,
      set_name_size = 4) + ggtitle("Jonathan Rotten")

v6 <- plot_grid(v1, v2, v3, v4, v5,
                nrow = 5, ncol = 1)
```

Make 4-way venn diagrams for timepoints, will drop one timepoint from each, limit of 4
```{r}
# Pulp - 4 timepoints
vjp1 <- ggvenn(dfj3, c("Pulp-Day 14", "Pulp-Day 28", "Pulp-Day 42", "Pulp-Rotten"),
      fill_color = c("#9ACA3C", "#40B9EB", "#426FB6", "#D771AD"),
      stroke_size = 0.5, 
      digits = 0,
      set_name_size = 4) + ggtitle("Jonathan Pulp")

vjp2 <- ggvenn(dfj3, c( "Pulp-Day 0",  "Pulp-Day 28", "Pulp-Day 42", "Pulp-Rotten"),
      fill_color = c("#ED1E24", "#40B9EB", "#426FB6", "#D771AD"),
      stroke_size = 0.5, 
      digits = 0,
      set_name_size = 4) + ggtitle("Jonathan Pulp")

vjp3 <- ggvenn(dfj3, c( "Pulp-Day 0", "Pulp-Day 14",  "Pulp-Day 42", "Pulp-Rotten"),
      fill_color = c("#ED1E24", "#9ACA3C", "#426FB6", "#D771AD"),
      stroke_size = 0.5, 
      digits = 0,
      set_name_size = 4) + ggtitle("Jonathan Pulp")
# Skin - 4 timepoints 
vjs1 <- ggvenn(dfj3, c("Skin-Day 14", "Skin-Day 28", "Skin-Day 42", "Skin-Rotten"),
      fill_color = c("#9ACA3C", "#40B9EB", "#426FB6", "#D771AD"),
      stroke_size = 0.5, 
      digits = 0,
      set_name_size = 4) + ggtitle("Jonathan Skin")

vjs2 <- ggvenn(dfj3, c( "Skin-Day 0",  "Skin-Day 28", "Skin-Day 42", "Skin-Rotten"),
      fill_color = c("#ED1E24", "#40B9EB", "#426FB6", "#D771AD"),
      stroke_size = 0.5, 
      digits = 0,
      set_name_size = 4) + ggtitle("Jonathan Skin")

vjs3 <- ggvenn(dfj3, c( "Skin-Day 0", "Skin-Day 14",  "Skin-Day 42", "Skin-Rotten"),
      fill_color = c("#ED1E24", "#9ACA3C", "#426FB6", "#D771AD"),
      stroke_size = 0.5, 
      digits = 0,
      set_name_size = 4) + ggtitle("Jonathan Skin")
```

### Gala
Repeat venn diagrams for Gala
```{r}
# Venn dataframe for Gala
dfg <- df4 %>%
  # extract categories from sample ID
  separate(col = "Sample_ID",
           into = c("Cultivar", "Treatment", "Tissue", "Timepoint", "Tree"),
           sep = "-",
           remove = FALSE) %>% 
  # subset by cultivar
  filter(Cultivar == "Gala") %>%
  # format variables by tissue and time
  group_by(Timepoint, Tissue) %>%
  summarise_if(is.numeric, sum) %>%
  ungroup() %>%
  mutate(Sample_ID = paste(Tissue, Timepoint, sep = "-")) %>%
  select(Sample_ID, starts_with("Otu"), -Tissue, -Timepoint)

dfg2 <- dfg %>%
  # transpose for venn format
  column_to_rownames("Sample_ID") %>%
  t() %>% as.data.frame() %>%
  # remove OTUs not found in any samples
  #rowSums() #	Otu59 = 0
  rownames_to_column(var = "Otu") %>%
  filter(Otu != "Otu59") %>% 
  # convert binary to logical
  mutate_if(is.numeric, as.logical)

v7 <- ggvenn(dfg2, c("Pulp-Day 0", "Skin-Day 0"),
      fill_color = c("#F9AA8F", "#ED1E24"),
      stroke_size = 0.5, 
      digits = 0,
      set_name_size = 4) + ggtitle("Gala Day 0")
v8 <- ggvenn(dfg2, c("Pulp-Day 14", "Skin-Day 14"),
      fill_color = c("#F9AA8F", "#ED1E24"),
      stroke_size = 0.5, 
      digits = 0,
      set_name_size = 4) + ggtitle("Gala Day 14")
v9 <- ggvenn(dfg2, c("Pulp-Day 28", "Skin-Day 28"),
      fill_color = c("#F9AA8F", "#ED1E24"),
      stroke_size = 0.5,
      digits = 0,
      set_name_size = 4) + ggtitle("Gala Day 28")
v10 <- ggvenn(dfg2, c("Pulp-Day 42", "Skin-Day 42"),
      fill_color = c("#F9AA8F", "#ED1E24"),
      stroke_size = 0.5, 
      digits = 0,
      set_name_size = 4) + ggtitle("Gala Day 42")
v11 <- ggvenn(dfg2, c("Pulp-Rotten", "Skin-Rotten"),
      fill_color = c("#F9AA8F", "#ED1E24"),
      stroke_size = 0.5, 
      digits = 0,
      set_name_size = 4) + ggtitle("Gala Rotten")

v12 <- plot_grid(v7, v8, v9, v10, v11,
                nrow = 5, ncol = 1) 

  # # change colors 
  # scale_color_manual(name = "Timepoint",
  #                    breaks = c("Day 0", "Day 14", "Day 28", 
  #                               "Day 42", "Rotten"),
  #                    values = c("#ED1E24", "#9ACA3C", "#40B9EB",
  #                               "#426FB6", "#D771AD")) 
```

Make 4-way venn diagrams for timepoints, will drop one timepoint from each, limit of 4
```{r}
# Pulp - 4 timepoints
vgp1 <- ggvenn(dfg2, c("Pulp-Day 14", "Pulp-Day 28", "Pulp-Day 42", "Pulp-Rotten"),
      fill_color = c("#9ACA3C", "#40B9EB", "#426FB6", "#D771AD"),
      stroke_size = 0.5, 
      digits = 0,
      set_name_size = 4) + ggtitle("Gala Pulp")

vgp2 <- ggvenn(dfg2, c( "Pulp-Day 0",  "Pulp-Day 28", "Pulp-Day 42", "Pulp-Rotten"),
      fill_color = c("#ED1E24", "#40B9EB", "#426FB6", "#D771AD"),
      stroke_size = 0.5, 
      digits = 0,
      set_name_size = 4) + ggtitle("Gala Pulp")

vgp3 <- ggvenn(dfg2, c( "Pulp-Day 0", "Pulp-Day 14",  "Pulp-Day 42", "Pulp-Rotten"),
      fill_color = c("#ED1E24", "#9ACA3C", "#426FB6", "#D771AD"),
      stroke_size = 0.5, 
      digits = 0,
      set_name_size = 4) + ggtitle("Gala Pulp")
# Skin - 4 timepoints 
vgs1 <- ggvenn(dfg2, c("Skin-Day 14", "Skin-Day 28", "Skin-Day 42", "Skin-Rotten"),
      fill_color = c("#9ACA3C", "#40B9EB", "#426FB6", "#D771AD"),
      stroke_size = 0.5, 
      digits = 0,
      set_name_size = 4) + ggtitle("Gala Skin")

vgs2 <- ggvenn(dfg2, c( "Skin-Day 0",  "Skin-Day 28", "Skin-Day 42", "Skin-Rotten"),
      fill_color = c("#ED1E24", "#40B9EB", "#426FB6", "#D771AD"),
      stroke_size = 0.5, 
      digits = 0,
      set_name_size = 4) + ggtitle("Gala Skin")

vgs3 <- ggvenn(dfg2, c( "Skin-Day 0", "Skin-Day 14",  "Skin-Day 42", "Skin-Rotten"),
      fill_color = c("#ED1E24", "#9ACA3C", "#426FB6", "#D771AD"),
      stroke_size = 0.5, 
      digits = 0,
      set_name_size = 4) + ggtitle("Gala Skin")
```
# Export
Data tables
```{r eval=FALSE, include=FALSE}
write_delim(shared_phylum, file = "shared_phylum.csv", delim = ",",
            col_names = TRUE, na = "NA")
write_delim(shared_class, file = "shared_class.csv", delim = ",",
            col_names = TRUE, na = "NA")
write_delim(shared_order, file = "shared_order.csv", delim = ",",
            col_names = TRUE, na = "NA")
write_delim(shared_family, file = "shared_family.csv", delim = ",",
            col_names = TRUE, na = "NA")
write_delim(shared_genus, file = "shared_genus.csv", delim = ",",
            col_names = TRUE, na = "NA")
```
Combined venn diagram figures
```{r}
save_plot(v6, filename = "Figures/venn_jonathan.pdf",
         base_height = 10, base_width = 4)
save_plot(v12, filename = "Figures/venn_gala.pdf",
         base_height = 10, base_width = 4)
save_plot(v18, filename = "Figures/venn_gala.pdf",
            base_height = 10, base_width = 4)
```
4-panel venn diagrams
```{r}
# Jonathan skin
save_plot(vjs1, filename = "Figures/venn_vjs1.pdf",
            base_height = 10, base_width = 4)
save_plot(vjs2, filename = "Figures/venn_vjs2.pdf",
            base_height = 10, base_width = 4)
save_plot(vjs3, filename = "Figures/venn_vjs3.pdf",
            base_height = 10, base_width = 4)
# Jonathan pulp
save_plot(vjp1, filename = "Figures/venn_vjp1.pdf",
            base_height = 10, base_width = 4)
save_plot(vjp2, filename = "Figures/venn_vjp2.pdf",
            base_height = 10, base_width = 4)
save_plot(vjp3, filename = "Figures/venn_vjp3.pdf",
            base_height = 10, base_width = 4)
# Gala skin
save_plot(vgs1, filename = "Figures/venn_vgs1.pdf",
            base_height = 10, base_width = 4)
save_plot(vgs2, filename = "Figures/venn_vgs2.pdf",
            base_height = 10, base_width = 4)
save_plot(vgs3, filename = "Figures/venn_vgs3.pdf",
            base_height = 10, base_width = 4)
# Gala pulp 
save_plot(vgp1, filename = "Figures/venn_vgp1.pdf",
            base_height = 10, base_width = 4)
save_plot(vgp2, filename = "Figures/venn_vgp2.pdf",
            base_height = 10, base_width = 4)
save_plot(vgp3, filename = "Figures/venn_vgp3.pdf",
            base_height = 10, base_width = 4)
```


# To-do
* plot replicates (5) within tree and between tree
* firmicutes in pulp from gala - where do these go? 
* group replicates make new version of plots for richness and shannon 
* make shared tables for excel 
* make genus-level barplot for actinobacteria, acidobacteria, bacteriodetes, firmicutes 
* jonathan venn diagram lists for excel 
```{r}

```

-----
end 
